framework/test/unit/modules/fee/fee_module.spec.ts

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright © 2021 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+
+import { FeeModule } from '../../../../src/modules/fee';
+import { VerifyStatus } from '../../../../src/node/state_machine';
+
+describe('FeeModule', () => {
+	let feeModule!: FeeModule;
+	let genesisConfig: any;
+	let moduleConfig: any;
+	let generatorConfig: any;
+
+	beforeEach(async () => {
+		genesisConfig = {
+			baseFees: [
+				{
+					commandID: 0,
+					baseFee: '1',
+					moduleID: 5,
+				},
+			],
+			minFeePerByte: 1000,
+		};
+		moduleConfig = {};
+		generatorConfig = {};
+		feeModule = new FeeModule();
+		await feeModule.init({ genesisConfig, moduleConfig, generatorConfig });
+	});
+
+	describe('constructor', () => {
+		it('should create instance of module', () => {
+			expect(feeModule).toBeInstanceOf(FeeModule);
+		});
+
+		it('should have valid id', () => {
+			expect(feeModule.id).toEqual(1);
+		});
+
+		it('should have valid name', () => {
+			expect(feeModule.name).toEqual('fee');
+		});
+	});
+
+	describe('init', () => {
+		it('should set the moduleConfig property', () => {
+			expect((feeModule as any)._moduleConfig).toEqual(moduleConfig);
+		});
+
+		it('should set the minFeePerByte property', () => {
+			expect((feeModule as any)._minFeePerByte).toEqual(1000);
+		});
+
+		it('should set the baseFees property', () => {
+			expect((feeModule as any)._baseFees).toEqual(
+				genesisConfig.baseFees.map((fee: any) => ({ ...fee, baseFee: BigInt(fee.baseFee) })),
+			);
+		});
+	});
+
+	describe('verifyTransaction', () => {
+		it('should validate transaction with sufficient min fee', async () => {
+			const transaction = {
+				moduleID: 5,
+				commandID: 0,
+				fee: BigInt(1000000000),
+				getBytes: () => Buffer.from('123456789'),
+			};
+			const result = await feeModule.verifyTransaction({ transaction } as any);
```

Comment:
Also, `createBlockContext` can be used to generate the input here

Reply:
Do you mean `createTransactionContext`? It does not return a type that matches with `TransactionVerifyContext`. Maybe it simpler to leave it as is.

Reply:
it would be better to consistently use the testing utils.
```
const context = createTransactionContext(tx);
const vcontext.createTransactionVerifyContext(); // this returns verify transaction context
```